intro.js

```diff
@@ -136,34 +137,40 @@
           currentItem.disableInteraction = this._options.disableInteraction;
         }
 
-        if (currentItem.element != null) {
+        if (currentItem.element !== null) {
           introItems.push(currentItem);
         }
       }
 
     } else {
       //use steps from data-* annotations
       var allIntroSteps = targetElm.querySelectorAll('*[data-intro]');
+      var elmsLength = allIntroSteps.length;
+      var currentElement;
+      var disableInteraction;
+
       //if there's no element to intro
-      if (allIntroSteps.length < 1) {
+      if (elmsLength < 1) {
         return false;
       }
 
+      i = 0;
+
       //first add intro items with data-step
-      for (var i = 0, elmsLength = allIntroSteps.length; i < elmsLength; i++) {
-        var currentElement = allIntroSteps[i];
+      for (; i < elmsLength; i++) {
+        currentElement = allIntroSteps[i];
 
         // skip hidden elements
-        if (currentElement.style.display == 'none') {
+        if (currentElement.style.display === 'none') {
           continue;
         }
 
         var step = parseInt(currentElement.getAttribute('data-step'), 10);
 
-        var disableInteraction = this._options.disableInteraction;
-
-        if (typeof (currentElement.getAttribute('data-disable-interaction')) != 'undefined') {
+        if (typeof (currentElement.getAttribute('data-disable-interaction')) !== 'undefined') {
           disableInteraction = !!currentElement.getAttribute('data-disable-interaction');
+        } else {
+          disableInteraction = this._options.disableInteraction;
```

Comment:
this is the base case and can be defined outside the if. like removing `var disableInteraction;` and initiating with  `this._options.disableInteraction`